parsers/IQ.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python3
+
+"""Parser for the electricity grid of Iraq"""
+
+import arrow
+import logging
+from dateutil.tz import tz
+import requests
+
+
+LIVE_PRODUCTION_API_URL = "https://www.gdoco.org/vueips.php"
+DATA_SOURCE = "www.gdoco.org"
+CELL_MAPPING = {
+    "hydro": ["1220"],
+    "gas": ["1219"],
+    "oil": [
+        "1218", # Thermal
+        "1221" # Diesel
+    ],
+}
+
+
+def template_response(zone_key, datetime, source):
+    return {
+        "zoneKey": zone_key,
+        "datetime": datetime,
+        "production": {
+            "gas": 0.0,
+            "hydro": 0.0,
+            "oil": 0.0,
+            "solar": None, # IEA has a count, but no data is available
+        },
+        "storage": {},
+        "source": source,
+    }
+
+def fetch_data(r):
+    resp = r.get(LIVE_PRODUCTION_API_URL)
+    data = resp.json()
+
+    timestamp = arrow.get(data["lastmodified"], "HH:mm:ss A DD-MM-YYYY", tzinfo=tz.gettz('Asia/Baghdad'))
+
+    return data["d"], timestamp
```

Comment:
As `datetime` must be returned as native `datetime.datetime` object

```suggestion
    return data["d"], timestamp.datetime
```